src/resolver.rs

```diff
@@ -809,6 +809,16 @@ impl<'i> TypeAnnotator<'i> {
                 } else if operator.is_bool_type() {
                     self.annotation_map
                         .annotate(statement, StatementAnnotation::value(BOOL_TYPE));
+                } else if left_type.get_type_information().is_pointer()
+                    || right_type.get_type_information().is_pointer()
+                {
+                    let target_name = if left_type.get_type_information().is_pointer() {
```

Comment:
what happens if left and right are both pointers?

Reply:
I think we have validation because that's not allowed.
If not we have to add it.
This is not ready btw, i'll mark it as draft it still does not work on &ref + 1